Project/Assets/ML-Agents/Examples/SharedAssets/Scripts/ModelCarousel.cs

```diff
@@ -0,0 +1,203 @@
+    }
+#endif
+
+    private void Reset()
+    {
+        m_Reset = false;
+        m_StepsSinceLastSwitch = 0;
+        m_CurrentModelIndex = 0;
+        m_Agent.SetModel(m_OriginalModel.name, m_OriginalModel);
+        textMeshComponent?.SetText("Ready to Start");
+    }
+
+    private void OnEnable()
+    {
+        m_Agent = GetComponent<Agent>();
+        m_OriginalModel = m_Agent.GetComponent<BehaviorParameters>().Model;
+
+        Reset();
+
+        k_FixedUpdatePerSecond = (int)(1.0f / Time.fixedDeltaTime);
+
+        if (m_TimeScaleOverride > 0.0f)
+        {
+            Time.timeScale = m_TimeScaleOverride;
+        }
+    }
+
+    void StartRecording()
+    {
+#if UNITY_EDITOR
+        if (!m_AutoRecord)
+            return;
+
+        Debug.Log("Starting Recording");
+        RecorderWindow recorderWindow = GetRecorderWindow();
+        if (!recorderWindow.IsRecording())
+            recorderWindow.StartRecording();
+#endif
+    }
+
+    void StopRecording()
+    {
+#if UNITY_EDITOR
+        if (!m_AutoRecord)
+            return;
+
+        Debug.Log("Stopping Recording");
+        RecorderWindow recorderWindow = GetRecorderWindow();
+        if (recorderWindow.IsRecording())
+            recorderWindow.StopRecording();
+#endif
+    }
+
+    void UpdateStepNumberText()
+    {
+        if (!m_ShowStepNumber)
+            return;
+
+        var result = Regex.Match(m_Models[m_CurrentModelIndex].name, @".*-(\d+)$");
+
+        string newText = "";
+        if (result.Success && result.Groups.Count > 0)
+        {
+            var steps = Int32.Parse(result.Groups[1].Captures[0].Value);
+
+            int round = m_StepNumberRounding;
+            steps += round / 2;
+            steps /= round;
+            steps *= round;
+
+            newText = $"After {steps:n0} steps";
+        }
+
+        textMeshComponent?.SetText(newText);
+    }
+
+    void SetModel()
+    {
+        if (m_CurrentModelIndex < 0 || m_CurrentModelIndex >= m_Models.Count)
+            return;
+
+        m_Agent.SetModel(m_Models[m_CurrentModelIndex].name, m_Models[m_CurrentModelIndex]);
+        m_CurrentlySetModelIndex = m_CurrentModelIndex;
+
+        UpdateStepNumberText();
+
+        if (m_ResetAgentOnModelChange)
+            m_Agent.EndEpisode();
+    }
+
+    void FixedUpdate()
+    {
+        if (m_Start)
+        {
+            m_Start = false;
+            m_Pause = false;
+            StartRecording();
+        }
+
+        if (m_Reset)
+        {
+            StopRecording();
+            Reset();
+            m_Pause = true;
+            m_Start = false;
+        }
+
+        if (m_Pause && !m_ForceNext)
+            return;
+
+        if (m_CurrentlySetModelIndex != m_CurrentModelIndex)
+        {
+            SetModel();
+        }
+
+        m_StepsSinceLastSwitch++;
+
+        if (m_StepsSinceLastSwitch >= m_SecondsBetweenSwitches * k_FixedUpdatePerSecond || m_ForceNext)
```

Comment:
`m_ForceNext` has not been set to `true` anywhere in the code. If it's related to `then use "Force Next" to advance.`, is there a place we need to set it to `true?

Reply:
Yes, there is. It's in the editor inspector window when you click on an object this has this script attached to it. This is a monobehaviour script, so any public property shows up in the inspector window for any object that has this script attached to it.